packages/application-shell/src/components/application-shell-authenticated/new-main-navigation-flag-wrapper.tsx

```diff
@@ -1,10 +1,25 @@
 // TODO: Remove this file once the new navigation is enabled for all users
-import { useFeatureToggle } from '@flopflip/react-broadcast';
+import { useFeatureToggle, useAdapterStatus } from '@flopflip/react-broadcast';
 import { MAIN_NAVIGATION } from '../../feature-toggles';
 
-const NewMainNavigationFlagWrapper = (props: { children: Function }) => {
+type TNewMainNavigationFlagWrapperChildrenProps = {
+  isNewNavigationEnabled: boolean;
+  isNewNavigationEnabledEvaluationPending: boolean;
+};
+
+const NewMainNavigationFlagWrapper = (props: {
+  children: ({
+    isNewNavigationEnabled,
+    isNewNavigationEnabledEvaluationPending,
+  }: TNewMainNavigationFlagWrapperChildrenProps) => JSX.Element;
+}) => {
   const isNewNavigationEnabled = useFeatureToggle(MAIN_NAVIGATION);
-  return props.children(isNewNavigationEnabled);
+  const { isConfiguring } = useAdapterStatus();
```

Comment:
I think you want `isReady` here. Ready state means the flags have been fetched. Configuration happens before.

Reply:
Thanks, I am 100% sure you are right, I need to have a proper look into it. I won't bother you before doing that.
Although it seems straightforward I simply couldn't make it work this way for now.

Just for some context:
[Here](https://github.com/commercetools/merchant-center-application-kit/compare/kk-navbar-skeleton..kk-navbar-skeleton-ff-evaluation) I pushed what supposedly should work.
But even thought the value of the flag is fetched, the hook's value of [adapterStatus](https://github.com/tdeekens/flopflip/blob/fd66f538640da02bdb6a3479a7a73135106cf142/packages/react-broadcast/src/hooks/use-adapter-status/use-adapter-status.ts#L14) strangely doesn't change.
The request:
<img width="1167" alt="Screenshot 2023-08-11 at 16 18 51" src="https://github.com/commercetools/merchant-center-application-kit/assets/49066275/ea12b46a-b008-4af5-80c8-5f7809d178d4">
The adapter's state is always `false`:
<img width="1053" alt="Screenshot 2023-08-11 at 16 27 29" src="https://github.com/commercetools/merchant-center-application-kit/assets/49066275/6d4b728a-b90b-43d6-96bb-d976c0ddbd80">

Reply:
Ok that's odd. I can see the same in the MC FE. Not sure if that's a bug in flopflip. Open to PRs there!

<img width="431" alt="CleanShot 2023-08-11 at 16 57 55@2x" src="https://github.com/commercetools/merchant-center-application-kit/assets/1877073/f503142f-d485-4aa3-ba27-a06ceb1b52f5">

Whatever works use it ;)

Reply:
> **Note:** a temporary workaround is used for computing the combined adapter status

https://github.com/commercetools/merchant-center-application-kit/blob/5eea6d29db452ecfa89994bcc47d8130e3c52515/packages/application-shell/src/components/application-shell-authenticated/new-main-navigation-flag-wrapper.tsx#L17-L20

Reply:
Let's see if this works. I can confirm that the combine-adapters doesn't combine the adapter status of the underlying adapters. We combine the launch-darkly and http-adapter to load flags from LD and the MC API. 

I can think of a way how to combine the adapter status without being ambiguous. Back then I opted out to wait until it would be needed.